### DISCUSSION TITLE
Docker basset storage:link not working

### DISCUSSION THREAD

**Discussion by Read2uk** [Laravel-Backpack/community-forum#1168] [Q&A (Help)] [answered]
hello all,

Just did the upgrade from backpack 5 to version 6.
All my assets are not loading because **php artisan storage:link** does not run on my docker environment when deploying to prod, so all the assets are 404.

I see basset added a post-install-cmd script in composer.json to run but on deployment I get this then it fails:

@php artisan storage:link --quiet
--
936 |  
937 | In Connection.php line 829:
938 |  
939 | SQLSTATE[HY000] [2002] Connection refused

</body>
</html>

What is the best way to fix this issue?

Is there a way to upgrade to backpack 6 without basset? Its causing me more troubles than its worth.

thank you all

**Comment by Read2uk** [comment]
also when running RUN composer run-script post-install-cmd:

> [8/8] RUN composer run-script post-install-cmd:
0.629 > @php artisan storage:link --quiet
0.787
0.795 In Application.php line 855:
0.795
0.795   Class "Backpack\Generators\GeneratorsServiceProvider" not found
0.795
0.795
0.801 Script @php artisan storage:link --quiet handling the post-install-cmd event returned with error code 1

**Comment by jcastroa87** [comment]
Hello @Read2uk 

What version of "backpack/generators" did you install?

Can you share all versions:
```
php artisan backpack:version
```

What kind of server and versions did you run in your docker container?

Cheers.

**Reply by Read2uk** [reply]
Hello @jcastroa87 , thank you for your help

### PHP VERSION:
8.3.12

### PHP EXTENSIONS:
Core, bcmath, calendar, ctype, date, filter, hash, iconv, json, SPL, pcre, random, readline, Reflection, session, standard, mysqlnd, tokenizer, zlib, libxml, dom, PDO, openssl, SimpleXML, xml, xmlreader, xmlwriter, curl, fileinfo, g
d, mbstring, Phar, pdo_mysql, zip

### LARAVEL VERSION:
11.25.0.0

### BACKPACK PACKAGE VERSIONS:
backpack/basset: 1.3.6
backpack/crud: 6.7.38
backpack/generators: v4.0.6
backpack/logmanager: v5.0.2
backpack/permissionmanager: 7.2.1
backpack/pro: 2.2.17
backpack/theme-coreuiv2: 1.2.5



Scripts in composer.json

    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-update-cmd": [
            "@php artisan vendor:publish --tag=laravel-assets --ansi --force"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ],
        "post-install-cmd": ["@php artisan storage:link --quiet"]
    },



Error when running the docker file:

 => CACHED [6/8] RUN composer install --no-dev --optimize-autoloader --no-scripts                                                                                                                                                  0.0s 
 => CACHED [7/8] RUN chmod -R 777 /var/www/html/storage/ /var/www/html/bootstrap/cache                                                                                                                                             0.0s 
 => ERROR [8/8] RUN composer run-script post-install-cmd                                                                                                                                                                           0.7s 
------
 > [8/8] RUN composer run-script post-install-cmd:
0.553 > @php artisan storage:link --quiet
0.633
0.639 In Application.php line 933:
0.639
0.639   Class "Backpack\Generators\GeneratorsServiceProvider" not found
0.639
0.639
0.643 Script @php artisan storage:link --quiet handling the post-install-cmd event returned with error code 1

**Reply by jcastroa87** [reply]
Try to find "GeneratorsServiceProvider" in your code.

I think as you try to update from v5 there is a file who try to call this class.

Please check this [old issue](https://github.com/Laravel-Backpack/community-forum/discussions/1000), maybe is the same for you.

Cheers.

**Reply by Read2uk** [reply]
Looks like very similar problem the one you mentioned.

Looks like if I delete the files from bootstrap/cache then run docker, it builds successfully .

I think its because when I run composer install locally, it installs the --dev scripts from composer.json and then creates the cache files with the dev packages, and when I run the docker file, it runs the composer install --no-dev meaning it won't have the packages on docker but the cache files already has references from dev packages.... 

Is this normal behaviour?

**Reply by Read2uk** [reply]
Also when it builds, looks like the post-install-cmd script to create the storage:link is not functioning properly as I have 404 on all assets from /storage

![image](https://github.com/user-attachments/assets/b1d91f9f-d90e-42d3-9413-a2737d4d69e3)

I feel like the basset its giving me a lot of problems, is there a way to run backpack v6 without the basset or /storage folder mapped to public?

**Reply by jcastroa87** [reply]
Your "APP_URL" in .env file is correctly set?

Then try to clean with:
```
php artisan basset:cache
php artisan basset:clear
```

Or check the issues with:
```
php artisan basset:check
```

About your question, basset is disabled by default in local dev, you can [check the docs](https://github.com/Laravel-Backpack/basset)

There are a lot of people who run Backpack in Docker without any issues, I think this issue is more connected to your server config or the way how you upgrade version.

Cheers.

**Reply by Read2uk** [reply, marked as answer]
Hi @jcastroa87 I kind of managed to get it working by adding this in the dockerfile:

RUN rm -rf /var/www/html/bootstrap/cache/*.php

Not sure if there is a better solution but this seems to work

thanks for your help
